product_harmonized_system/models/product_product.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2011-2022 Akretion (http://www.akretion.com)
+# Copyright 2009-2022 Noviat (http://www.noviat.com)
+# @author Alexis de Lattre <alexis.delattre@akretion.com>
+# @author Luc de Meyer <info@noviat.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def get_hs_code_recursively(self):
+        res = self.env["hs.code"]
+        if self:
```

Comment:
Could we really have `self` not set ?
Maybe it is a good time to add some quick unit tests for this module although there is not much code in it.

Reply:
Maybe not need to check `self` not set? It's away setted when call by `product_id`